docs/fsharp/tutorials/asynchronous-and-concurrent-programming/async.md

```diff
@@ -1,213 +1,349 @@
 ---
-title: Async Programming
-description: Learn how F# async programming is accomplished via a language-level programming model that is easy to use and natural to the language.
-ms.date: 06/20/2016
+title: Async Programming in F#
+description: Learn how F# async programming is accomplished via a language-level programming model that utilizes core functional programming concepts to allow for easy use of asynchrony in F# programming.
+ms.date: 12/17/2018
 ---
 # Async Programming in F# #
 
-> [!NOTE]
-> Some inaccuracies have been discovered in this article.  It is being rewritten.  See [Issue #666](https://github.com/dotnet/docs/issues/666) to learn about the changes.
+Asynchronous programming is essential to modern applications. The applications for asynchronous programming are large in number. There are two primary use cases that most developers will encounter:
 
-Async programming in F# can be accomplished through a language-level programming model designed to be easy to use and natural to the language.
+* Having a non-blocking server process that can service incoming requests while awaiting results for other requests
+* Having a responsive UI or main thread while background work is happening
 
-The core of async programming in F# is `Async<'T>`, a representation of work that can be triggered to run in the background, where `'T` is either the type returned via the special `return` keyword or `unit` if the async workflow has no result to return.
+Although background work may involve multiple threads, there is no affinity between threading and asynchrony.
 
-The key concept to understand is that an async expression’s type is `Async<'T>`, which is merely a _specification_ of work to be done in an asynchronous context. It is not executed until you explicitly start it with one of the starting functions (such as `Async.RunSynchronously`). Although this is a different way of thinking about doing work, it ends up being quite simple in practice.
+## Asynchrony defined
 
-For example, say you wanted to download the HTML from dotnetfoundation.org without blocking the main thread. You can accomplish it like this:
+If you look at the etymology of the word "asynchronous", there are two pieces:
+
+* "a", meaning "not"
+* "synchronous", meaning "at the same time"
+
+Put together, "asynchronous" simply means, "not at the same time". This holds true for asynchronous programming in F#. Code that executes asynchronously is code that does not necessarily execute at the same time.
+
+In other words, asynchronous computations execute independently of the main program flow. This does not imply any particular threading model, nor does it imply that a computation happens in the background. In fact, asynchronous computations could even execute synchronously, depending on the nature of the computation and the environment the computation is executing in.
```

Comment:
>execute independently of the main program flow
this is far too concrete for this point in the explanation - I'm thinking folks have tied it down to there being exactly two threads operating in some very closely controlled interlocked dance now.

The next sentence untangles it a bit, but the strawman has already done the damage IMO

>could even execute synchronously

may even be executed synchronously inline ?

Maybe extend the concurrency etymological exploration by explaining that concurrency can be abstract - it affords one the ability to defer the exact time and manner in which to actually run things a) on a thread b) at a particular point in time c) with a particular priority relative to other work ?